cloud_storage/onedrive/upload_file.py

```diff
@@ -0,0 +1,231 @@
+    return destination_file_name
+
+
+def determine_destination_file_name(
+    *,
+    source_full_path,
+    destination_file_name,
+        file_number=None):
+    """
+    Determine if the destination_file_name was provided, or should be extracted
+    from the source_file_name, or should be enumerated for multiple file
+    uploads.
+    """
+    if destination_file_name:
+        if file_number:
+            destination_file_name = enumerate_destination_file_name(
+                destination_file_name, file_number)
+        else:
+            destination_file_name = destination_file_name
+    else:
+        destination_file_name = extract_file_name_from_source_full_path(
+            source_full_path)
+
+    return destination_file_name
+
+
+def clean_folder_name(folder_name):
+    """
+    Cleans folders name by removing duplicate '/' as well as leading and
+    trailing '/' characters.
+    """
+    folder_name = folder_name.strip('/')
+    if folder_name != '':
+        folder_name = os.path.normpath(folder_name)
+    return folder_name
+
+
+def combine_folder_and_file_name(folder_name, file_name):
+    """
+    Combine together the provided folder_name and file_name into one path
+    variable.
+    """
+    combined_name = os.path.normpath(
+        f'{folder_name}{"/" if folder_name else ""}{file_name}')
+    combined_name = os.path.normpath(combined_name)
+
+    return combined_name
+
+
+def determine_destination_full_path(
+        destination_folder_name,
+        destination_file_name,
+        source_full_path,
+        file_number=None):
+    """
+    Determine the final destination name of the file being uploaded.
+    """
+    destination_file_name = determine_destination_file_name(
+        destination_file_name=destination_file_name,
+        source_full_path=source_full_path,
+        file_number=file_number)
+    destination_full_path = combine_folder_and_file_name(
+        destination_folder_name, destination_file_name)
+    return destination_full_path
+
+
+def find_all_local_file_names(source_folder_name):
+    """
+    Returns a list of all files that exist in the current working directory,
+    filtered by source_folder_name if provided.
+    """
+    cwd = os.getcwd()
+    cwd_extension = os.path.normpath(f'{cwd}/{source_folder_name}/**')
+    file_names = glob.glob(cwd_extension, recursive=True)
+    return file_names
+
+
+def find_all_file_matches(file_names, file_name_re):
+    """
+    Return a list of all file_names that matched the regular expression.
+    """
+    matching_file_names = []
+    for file in file_names:
+        if re.search(file_name_re, file):
+            matching_file_names.append(file)
+
+    return matching_file_names
+
+
+def upload_onedrive_file(
+        client,
+        source_full_path,
+        destination_full_path):
+    """
+    Uploads a single file to OneDrive.
+    """
+    destination_path, destination_name = destination_full_path.rsplit('/', 1) 
```

Comment:
Is `destination_full_path` guaranteed to have a `/` in it. If it doesn't, wouldn't `rsplit` come back with just one element, not the two that are expected?